provider/fs/src/fs_data_provider.rs

```diff
@@ -146,3 +147,45 @@ where
         self.as_deserializing().load_payload(key, req)
     }
 }
+
+impl IterableProvider for FsDataProvider {
+    fn supported_options_for_key(
```

Comment:
It's just that the code got pretty messy because there are conversions between `OsStr`, `Path`, and `str` and unwraps all over the place :(

For `FsProvider` this feels a lot more brittle than for `BlobProvider` anyway.